tensorboard/webapp/hparams/_redux/hparams_data_source.ts

```diff
@@ -143,40 +162,35 @@ export class HparamsDataSource {
         'request'
       )
       .pipe(
-        map((sessionGroupsList) => {
-          const runToHparamsAndMetrics: RunToHparamsAndMetrics = {};
-
-          // Reorganize the sessionGroup/session into run to <hparams,
-          // metrics>.
-          for (const sessionGroup of sessionGroupsList.sessionGroups) {
-            const hparams: HparamValue[] = Object.entries(
-              sessionGroup.hparams
-            ).map((keyValue) => {
-              const [hparam, value] = keyValue;
-              return {name: hparam, value: value as DiscreteHparamValue};
-            });
-
-            for (const session of sessionGroup.sessions) {
-              for (const metricValue of session.metricValues) {
-                const runName = metricValue.name.group
-                  ? `${session.name}/${metricValue.name.group}`
-                  : session.name;
-                const runId = `${formattedExperimentIds}/${runName}`;
-                const hparamsAndMetrics = runToHparamsAndMetrics[runId] || {
-                  metrics: [],
-                  hparams,
-                };
-                hparamsAndMetrics.metrics.push({
-                  tag: metricValue.name.tag,
-                  trainingStep: metricValue.trainingStep,
-                  value: metricValue.value,
-                });
-                runToHparamsAndMetrics[runId] = hparamsAndMetrics;
+        map((response) =>
+          response.sessionGroups.map((sessionGroup) => {
+            sessionGroup.sessions = sessionGroup.sessions.map((session) => {
+              /**
+               * In single experiment mode the Session.name is equal to the runName.
+               * In comparison view it is `[AliasNumber] ExperimentAlias/runName`
+               *
+               * We store runs as experimentId/runName so it is necessary to prepend the experiment name
+               * in single experiment view. In comparison view we pass the index of the experimentId as the alias allowing
```

Comment:
This last sentence is sort of run-on and imprecise. Consider:

"In comparison view we pass the indeces of the experimentIds as the aliases in the request. That allows us to parse the indeces from the response and use them to lookup the correct ids from the experimentIds argument."